client/src/app/site/agenda/components/agenda-list/agenda-list.component.ts

```diff
@@ -341,4 +342,40 @@ export class AgendaListComponent extends BaseListViewComponent<ViewItem> impleme
             return result;
         }
     }
+
+    /**
+     * Duplicates a single selected item.
+     *
+     * @param item The item to duplicte.
+     */
+    public duplicateSingleTopic(item: ViewItem): void {
+        const topic = item.contentObject.getModel() as Topic;
+        this.topicRepo.create(
+            new Topic({ text: topic.text, title: `${topic.title} - ${this.translate.instant('duplicate')}` })
+        );
+    }
+
+    /**
+     * Duplicates all selected items, that are topics.
+     *
+     * @param selectedItems The items to duplicate.
+     */
+    public duplicateMultipleTopics(selectedItems: ViewItem[]): void {
+        for (const item of selectedItems) {
+            if (this.isTopic(item)) {
+                this.duplicateSingleTopic(item);
+            }
+        }
+    }
```

Comment:
Same applies as above, while you would need a try-catch block if the function above throws an error